File: tests/go/cases/case_143_closure_chain_async_05.go
package escape_tests

var retainedCase143 = []map[string]string{}

func Case143ClosureChainAsync05(input string) string {
	raw := input
	if raw == "" {
		raw = "sample"
	}
	payload := map[string]string{
		"task": "closure_chain_async_05",
		"entity": "stress",
		"stage": "evaluation",
		"input": raw,
		"checksum": "closure_chain_async_05:" + raw,
	}
	handler := func() string { return payload["input"] }
	_ = handler
	// ESCAPE: closure payload persisted indirectly via retained metadata map.
	retainedCase143 = append(retainedCase143, map[string]string{"h": payload["input"]})
	return "ok"
}

File: tests/nodejs/cases/case_143_closure_chain_async_05.js
/**
 * Task case 143: closure_chain_async_05 false-positive/false-negative stress pattern.
 */

const retainedCache = new Map();
const retainedAudit = [];
const retainedHandlers = [];

function case143ClosureChainAsync05(input) {
  const taskName = 'closure_chain_async_05';
  const raw = input || 'sample';
  const payload = {
    task: taskName,
    entity: 'stress',
    stage: 'evaluation',
    input: raw,
    checksum: `${taskName}:${raw.length}`,
  };
  const handler = () => payload.input;
  // ESCAPE: retained closure captures payload.
  retainedHandlers.push(handler);
  return 'ok';
}

module.exports = {
  case143ClosureChainAsync05,
};
